open-api/rest-catalog-open-api.yaml

```diff
@@ -2068,6 +2162,145 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        A JSON object that contains information provided by the server,
+        to be utilized by clients for distributed planning, should be supplied
+        as is for input in PlanTable operation.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - schema
+        - spec
+        - start
+        - length
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/ContentFile'
+        partition:
+          type: object
+          additionalProperties:
+            type: string
+        size-bytes:
+          type: number
+        start:
+          type: number
+        length:
+          type: number
+        estimated-rows-count:
+          type: number
+        delete-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/ContentFile'
+        schema:
+          $ref: '#/components/schemas/Schema'
```

Comment:
What's the purpose of sending a schema? What schema is this, the table schema, read schema, data file schema, or partition schema?

Reply:
When modeling this, I referred to the `FileScanTaskParser`
https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/FileScanTaskParser.java , which contains a field `SCHEMA`
```
  private static final String SCHEMA = "schema";
```
This seems to be linked to a Table Schema, since the BaseFileScanTask has a reference to this Schema class 

```
/**
 * The schema of a data table.
 *
 * <p>Schema ID will only be populated when reading from/writing to table metadata, otherwise it
 * will be default to 0.
 */
public class Schema implements Serializable {

```
https://github.com/apache/iceberg/blob/5b84f34a5386fc61b17bfe7dc7c1cbe565550958/api/src/main/java/org/apache/iceberg/Schema.java#L51

Reply:
yes, schema is needed to serialize file scan task
```
  public BaseFileScanTask(
      DataFile file,
      DeleteFile[] deletes,
      String schemaString,
      String specString,
      ResidualEvaluator residuals) 
```

Reply:
Looks like this is the schema of the partition spec of the manifest file, which means this is the default schema of the table that is bounded to the spec when initializing the table?

At least this seems to be what these code snippets suggest:

```
ManifestGroup manifestGroup =
        new ManifestGroup(io, dataManifests, deleteManifests)
            .caseSensitive(isCaseSensitive())
            .select(scanColumns())
            .filterData(filter())
            .specsById(table().specs())
            .scanMetrics(scanMetrics())
            .ignoreDeleted();
```

```
LoadingCache<Integer, TaskContext> taskContextCache =
        Caffeine.newBuilder()
            .build(
                specId -> {
                  PartitionSpec spec = specsById.get(specId);
                  ResidualEvaluator residuals = residualCache.get(specId);
                  return new TaskContext(spec, deleteFiles, residuals, dropStats, scanMetrics);
                });
```

```
TaskContext(
        PartitionSpec spec,
        DeleteFileIndex deletes,
        ResidualEvaluator residuals,
        boolean dropStats,
        ScanMetrics scanMetrics) {
      this.schemaAsString = SchemaParser.toJson(spec.schema());
      this.specAsString = PartitionSpecParser.toJson(spec);
      this.deletes = deletes;
      this.residuals = residuals;
      this.dropStats = dropStats;
      this.scanMetrics = scanMetrics;
    }
```

In that case it seems like it can be omitted.

Reply:
Will remove it

Reply:
Yeah, if this is based on what gets serialized to workers in Java frameworks, we don't need it. Those tasks send the table schema and spec to be able to work with the partition tuple on the task side. But this use case assumes that the caller has access to the table.

If we wanted it to be possible for the caller to not load the table (which we may choose to do in a later update to this API) then we would send this metadata once per request rather than on each task.